src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2071,7 +2071,14 @@ def construct_command(self, task_exec, task_args, task_numcores,
             task_command = task_exec
 
         # Construct the hosts_string, env vars
-        hosts_string = ",".join([slot.split(':')[0] for slot in task_slots])
+        # On some Crays, like on ARCHER, the hostname is "archer_N".
+        # In that case we strip off the part upto and including the underscore.
+        #
+        # TODO: If this ever becomes a problem, i.e. we encounter "real" hostnames
+        #       with underscores in it, or other hostname mangling, we need to turn
+        #       this into a system specific regexp or so.
+        #
```

Comment:
Maybe I still don't really understand what this is supposed to do, but I though this would translate hostname aliases as given by the nodelist into hostnames which are usable to connect to the hosts on IP/TCP level, ie. valid hostnames.  One would think that the things given in the hostlist would be translatable via the 'host' command?